Tests/Recurly/Base_Test.php

```diff
@@ -63,4 +63,27 @@ public function testCheckIfResponseContainGetHeadersFunction() {
     $this->assertTrue(method_exists($adjustment, 'getHeaders'),'Adjustments Class does not have method getHeaders');
     $this->assertInternalType('array', $adjustment->getHeaders());
   }
+
+  public function testPassingEmptyResourceCode() {
+    $this->client->addResponse('GET', '/accounts/abcdef1234567890', 'accounts/show-200.xml');
+    try {
+      $uri = Recurly_Base::_safeUri(Recurly_Client::PATH_ACCOUNTS, "");
+      Recurly_Base::_get($uri, $this->client);
+      $this->fail("Expected Recurly_Error");  
+    } catch (Recurly_Error $e) {
+      $this->assertEquals('Resource code cannot be an empty value', $e->getMessage());
+    }
+
+    $this->client->addResponse('GET', '/subscriptions/012345678901234567890123456789ab/add_ons/marketing_emails/usage/123456', 'usage/show-200.xml');
+    try {
+      $uri = Recurly_Base::_safeUri(
+        Recurly_Client::PATH_SUBSCRIPTIONS, "", 
+        Recurly_Client::PATH_ADDONS, "marketing_emails",
```

Comment:
If all these start with a `/`, how is that coming out in the url? your function doesn't seem to check for it. so it's either gonna be double `/` or a single `/` and an encoded one.